resources/packs/core/core_test.go

```diff
@@ -1057,6 +1061,16 @@ func TestListResource(t *testing.T) {
 			"users.map(name)",
 			0, []interface{}([]interface{}{"root", "chris", "christopher", "chris", "bin"}),
 		},
+		{
+			// outside variables cause the block to be standalone
+			"n=false; users.contains(n)",
+			1, false,
+		},
+		{
+			// variables do not override local fields in blocks
+			"name=false; users.contains(name)",
```

Comment:
I do not understand, why this test passes.

When I run this on my local Linux machine, I also get an "ok". But I would expect something else:
```
cnquery> name=false; users.contains(name)
[ok] value: 56
```
"56" is the length of the users array. But what does it check against, that it returns "ok"? The last element in the users array is:
```
55: user name="snap_daemon" uid=584788 gid=584788
```

When I change the query, I get the result I'd expect:
```
cnquery> name=false; users.map(name).contains(name)
[failed] [].contains()
  expected: > 0
  actual:   0
```

Reply:
It seems that this issue happens when the slice is a list of resources. We probably need to handle this separately. Lets get this merged anyway since it fixes an active issue.

Reply:
@czunker Actually this is behaving correctly right now: it uses `user.name` inside the block, instead of the variable you declared. That's because the variable is overwritten by the block's own fields. However, we may want to change this behavior, and feedback very warmly welcome on this! I can see how this is confusing and we may want the outer variable to win this race, instead of the field. Users can still address the field via `_`.

The best way to see it is to try this:

```coffee
> name="root"; users.contains(name == "root")
[ok] value: 1
```

The resulting list is 1, because it only applies to 1 name ^^

```coffee
> name2="root"; users.contains(name2 == "root")
[ok] value: 37
```

The resulting list is 37, because it now applies to everything, because it uses the variable

Reply:
I completely have not thought about that we overshadow the variable here. I think it is standard that the deepest level overrides the level above. Instead of fixing the problem, only to discover than then some users need exact the opposite, I recommend we add a compiler warning to cases where we see variable shadowing happening. This would be similar to `go build` and `go vet`

Reply:
Thanks for the explanation. That helped to understand what happens here.